bindings/python/src/lib.rs

```diff
@@ -220,15 +220,27 @@ enum Device {
     Cpu,
     Cuda(usize),
     Mps,
+    Npu(usize),
 }
 
 impl<'source> FromPyObject<'source> for Device {
     fn extract(ob: &'source PyAny) -> PyResult<Self> {
-        if let Ok(name) = ob.extract::<String>() {
+        let name = if let Ok(name) = ob.extract::<String>() {
+            Ok(name)
+        } else {
+            Python::with_gil(|py| {
+                ob.getattr(intern!(py, "__str__"))?
+                    .call0()?
+                    .extract::<String>()
+            })
+        };
```

Comment:
Why do we need this ? Can't we remove it ?

Calling `__str__` directly feels too much like a hack.